snap/snapcraft.yaml

```diff
@@ -21,26 +24,71 @@ apps:
       - optical-drive
       - removable-media
     common-id: dev.linwood.butterfly
-    desktop: usr/share/applications/dev.linwood.butterfly.desktop
+    desktop: meta/gui/butterfly.desktop
 
 parts:
+  flutter-git:
+    source: https://github.com/flutter/flutter.git
+    source-tag: 3.10.1
+    plugin: nil
+    override-build: |
+      set -eux
+      mkdir -p $CRAFT_PART_INSTALL/usr/bin
+      mkdir -p $CRAFT_PART_INSTALL/usr/libexec
+      cp -r $CRAFT_PART_SRC $CRAFT_PART_INSTALL/usr/libexec/flutter
+      ln -sf $CRAFT_PART_INSTALL/usr/libexec/flutter/bin/flutter $CRAFT_PART_INSTALL/usr/bin/flutter
+      export PATH="$CRAFT_PART_INSTALL/usr/bin:$PATH"
+      flutter doctor
+    build-packages:
+      - clang
+      - cmake
+      - curl
+      - ninja-build
+      - unzip
+    override-prime: ''
   butterfly:
-    plugin: dump
+    after: [ flutter-git ]
+    plugin: nil
+    #plugin: flutter
     source: https://github.com/LinwoodDev/Butterfly.git
+    source-tag: v2.0.0-alpha.3
```

Comment:
Why hardcoding source-tag?
I would always try to build the current tag

Reply:
That's not hardcode, you can do it in two ways, only updating with every new `source-tag`, or directly pushing every new commit in edge channel as a nightly package. Again please read the commit messages.

>There is one more thing, that can be done, that's using the repo directly as source, without any source-tag. Thus will allow to make nightly snaps, pushed to edge channel.

Reply:
Yeah I would do that instead, I have a `nightly` were every new release will be pushed